qiime2_pipeline.bash

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
+
+if [ $# -ne 4 ]; then
+	echo "Usage: $0 WORK_DIR MAPPING_FP LANE_NUM PROJECT_NAME"
+	exit 1
+fi
+
+WORK_DIR=$1
+MAPPING_FP=$2
+LANE_NUM=$3
+PROJECT_NAME=$4
+
+## GreenGene Training Set artiffact FP
+## This can be downloaded from https://data.qiime2.org/2017.8/common/gg-13-8-99-nb-classifier.qza
+gg_FP=$HOME"/gg-13-8-99-nb-classifier.qza"
+
+DATA_DIR=$WORK_DIR"/data_files"
+
+PROJECT_DIR=$WORK_DIR"/"$PROJECT_NAME
+if [ ! -d $PROJECT_DIR ]; then
+	mkdir $PROJECT_DIR
+fi
+
+FWD=$DATA_DIR"/Undetermined_S0_L$(printf "%03d" $LANE_NUM)_R1_001.fastq.gz"
+REV=$DATA_DIR"/Undetermined_S0_L$(printf "%03d" $LANE_NUM)_R2_001.fastq.gz"
+IDX=$DATA_DIR"/Undetermined_S0_L$(printf "%03d" $LANE_NUM)_I12_001.fastq.gz"
+
+###=====================
+### DATA IMPORT 
+###=====================
+
+EMP_PAIRED_END_SEQUENCES_DIR=$PROJECT_DIR"/emp-paired-end-sequences"
+if [ ! -d $EMP_PAIRED_END_SEQUENCES_DIR ]; then
+        mkdir $EMP_PAIRED_END_SEQUENCES_DIR
+fi
+
+cp $FWD $EMP_PAIRED_END_SEQUENCES_DIR/"forward.fastq.gz"
+cp $REV $EMP_PAIRED_END_SEQUENCES_DIR/"reverse.fastq.gz"
+cp $IDX $EMP_PAIRED_END_SEQUENCES_DIR/"barcodes.fastq.gz"
+
+qiime tools import \
+  --type EMPPairedEndSequences \
+  --input-path $EMP_PAIRED_END_SEQUENCES_DIR \
+  --output-path $PROJECT_DIR"/emp-paired-end-sequences.qza"
+
+###=====================
+### DEMULTIPLEXING SEQUENCE
+###=====================
+
+qiime demux emp-paired \
+  --m-barcodes-file $MAPPING_FP \
+  --m-barcodes-category BarcodeSequence \
+  --i-seqs $PROJECT_DIR"/emp-paired-end-sequences.qza" \
+  --o-per-sample-sequences $PROJECT_DIR"/demux" \
```

Comment:
Is this equivalent to `demux.qza`?

Reply:
The output from DEMULTIPLEXING SEQUENCE step is demux.qza

Reply:
--p-rev-comp-mapping-barcodes is followed by the lines above. Depending on how we create BarcodeSequence in the mapping file, this flag may not be necessary.